lib/ret.ex

```diff
@@ -16,4 +21,76 @@ defmodule Ret do
       {:error, :forbidden}
     end
   end
+
+  defp delete_avatars_for_account(%Account{} = account) do
+    account_avatars_query = from(avatar in Avatar, where: avatar.account_id == ^account.account_id)
+    account_avatars = Repo.all(account_avatars_query) |> Repo.preload(Avatar.file_columns())
+
+    account_avatar_owned_files =
+      account_avatars
+      |> Enum.flat_map(fn avatar ->
```

Comment:
Does this actually need to be flattened? The variable name in `Enum.filter/2` would seem to indicate that it does not.

Reply:
Hmm. Each avatar has multiple OwnedFiles associated with it. This iterates through each avatar, fetches each OwnedFile, filters the nil ones, and then flattens them such that we have a single array of all the OwnedFiles associated with all the avatars belonging to the user. So yeah, it does need to be flattened.